services/business/checkers/virtual_service_checker.go

```diff
@@ -0,0 +1,71 @@
+package checkers
+
+import (
+	"github.com/kiali/kiali/kubernetes"
+	"github.com/kiali/kiali/services/business/checkers/virtual_services"
+	"github.com/kiali/kiali/services/models"
+)
+
+const virtualCheckerType = "virtualservice"
+
+type VirtualServiceChecker struct {
+	Namespace        string
+	DestinationRules []kubernetes.IstioObject
+	VirtualService   []kubernetes.IstioObject
+}
+
+// An Object Checker runs all checkers for an specific object type (i.e.: pod, route rule,...)
+// It run two kinds of checkers:
+// 1. Individual checks: validating individual objects.
+// 2. Group checks: validating behaviour between configurations.
+func (in VirtualServiceChecker) Check() models.IstioValidations {
+	validations := models.IstioValidations{}
+
+	validations = validations.MergeValidations(in.runIndividualChecks())
+	validations = validations.MergeValidations(in.runGroupChecks())
+
+	return validations
+}
+
+// Runs individual checks for each virtual service
+func (in VirtualServiceChecker) runIndividualChecks() models.IstioValidations {
+	validations := models.IstioValidations{}
+
+	for _, virtualService := range in.VirtualService {
+		in.runChecks(virtualService, validations)
+	}
+
+	return validations
+}
+
+// runGroupChecks runs group checks for all virtual services
+func (in VirtualServiceChecker) runGroupChecks() models.IstioValidations {
+	return models.IstioValidations{}
+}
+
+// enabledCheckersFor returns the list of all individual enabled checkers.
+func (in VirtualServiceChecker) enabledCheckersFor(object kubernetes.IstioObject) []Checker {
```

Comment:
I wouldn't use a function for a one-line assignment :-).
Sorry, it sounds over-splitting. 
I guess a function should perform at least two operations.